### ISSUE TITLE
cli: list --show-deleted-runs

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana-client#612]
**Current behaviour**

The `reana-client` is showing all runs by default, included deleted ones:

```console
$ reana-client list
NAME                                RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS
root6-roofit-yadage-kubernetes      1            2022-06-01T15:07:49   2022-06-01T15:08:02   2022-06-01T15:09:10   finished
root6-roofit-snakemake-kubernetes   1            2022-06-01T15:06:58   2022-06-01T15:07:11   2022-06-01T15:07:41   deleted
root6-roofit-serial-kubernetes      1            2022-06-01T15:06:18   2022-06-01T15:06:30   2022-06-01T15:06:49   finished
root6-roofit-cwl-kubernetes         1            2022-06-01T15:05:37   2022-06-01T15:05:55   2022-06-01T15:06:11   finished
```

**Expected behaviour**

In https://github.com/reanahub/reana-ui/issues/218 we are switching to exclude deleted runs from Web UI by default:

![show-deleted-runs](https://user-images.githubusercontent.com/517546/171824706-8f1015cd-f366-4134-bca5-c84f81e2d547.png)

The toggle will be "off" by default and deleted runs won't be shown by default to the Web users.

This issue is about doing the same change for CLI users.

We can achieve the same UX by introducing a new option for the `list` command:

```console
$ reana-client list --show-deleted-runs
```

so that Web UI and CLI UI would behave fully consistently.

**Notes**

**(1)** Note that the new option `--show-deleted-runs` would work orthogonally alongside `--include-progress` and `--include-workspace-size`  options, since `--show-...` would operate on rows and `-include-...` would operate on columns of the output matrix.

**(2)** Note that we may need to take care about `--filter status=deleted` and how this would play with `--show-deleted-runs`. In Web UI, we solved it by not allowing the former until the latter is flipped. We can do the same here.

**(3)** Note that this is seemingly bringing the old `reana-client` behaviour back, before we switched to showing deleted runs by default, see https://github.com/reanahub/reana-ui/issues/218 for the past motivations. However, I think that these concerns can be addressed by doing two more changes:

- in CLI, `reana-client delete` would include by default also the workspace.
- in Web UI, we won't allow to "untick" option to preserve workspace in the following screenshot:

![delete-also](https://user-images.githubusercontent.com/517546/171824240-d026f80b-1b52-409b-aca7-36d1e7993273.png)

In this way, the deletion of a workflow runs would always trigger the deletion of its workspace. If we give enough warnings to the user prior to performing deletions, I think that this should do. I cannot think of a use case where one would like to "delete" a run but "preserve" its workspace; it seems contradictory. I think we can afford removing this option.

**Comment by VMois** [comment]
To confirm behavior:

```bash
$ reana-client list  # doesn't show deleted workflows
$ reana-client list --show-deleted-runs  # shows deleted workflows
$ reana-client list --filter status=deleted  # doesn't show deleted workflows, displays warning to user to use "--show-deleted-runs" option
$ reana-client list --filter status=deleted --show-deleted-runs  # shows deleted workflows
$ reana-client delete  # deletes workflow AND its workspace
```

Is it correct @tiborsimko

**Comment by tiborsimko** [comment]
@VMois Yes, except one minor precision; the following bit:

```console
$ reana-client list --filter status=deleted --show-deleted-runs  # shows deleted workflows
```

will show _only_ deleted runs.

**Comment by VMois** [comment]
In `reana-client`, we already have the `--all` parameter that is by default `True`. It allows displaying deleted workflows. A very simple solution is to rename it to `--show-deleted-runs` and set the default to `False`.

It can create a performance issue if you have a lot of deleted workflows, so probably, we need a more optimized way.

**Comment by VMois** [comment]
I have a few questions:

1. In the example below, when `--show-deleted-runs` is included I kinda expect deleted workflows to be displayed. Cause option says `show`. It might be confusing to the user. Should we, maybe, include deleted workflows whatever `show-deleted-runs` is used?

```bash 
$ reana-client list --filter status=finished  # show "finished" workflows only, do not show deleted workflows
$ reana-client list --filter status=finished  --show-deleted-runs  # show "finished" workflows only, do not show deleted workflows (the same behavior as in the line above)

2. Should I fix the `all` option? Currently, whether it is used or not the output is the same.

3. The example below sounds quite unnecessary. Maybe, if `filter=deleted` is used, we should just display deleted workflows and not ask for `show-deleted-runs`. If someone types `filter=deleted` they are pretty much aware of what they want. WDYT?

```bash
$ reana-client list --filter status=deleted  # warns with message to include "--show-deleted-runs"
$ reana-client list --filter status=deleted  --show-deleted-runs  # show only deleted workflows 
```

**Comment by tiborsimko** [comment]
WRT 1, when a user uses `--show-deleted-runs` toggle, the client should display them, but only if it matches other constrains.  If you ask for `--filter name=myfooanalysis` and deleted runs are all named `mybaranalysis`, then nothing would be shown.  

(Basically, the name of the option means "show also deleted runs if they are matched". If you think it would be clearer to call the option differently, perhaps "include deleted runs" or some such, we can ponder renaming... But let's use consistent names in Web UI and CLI UI whatever we choose.

WRT 2, when I used `--all`, I get also deleted runs. So it seems to do what the docstring says.  It's basically just a backwards-compatibility thing.  The new behaviour would be: hide deleted runs unless `--show-deleted-runs` or -all` is used.

WRT 3, yes that is a good point. Choosing `--filter deleted` would automatically enable option `--show-deleted-runs`, so to speak. Very logical.  The Web UI behaves differently, because it does not show "deleted"  as an available value in the Status drop-down. But, for CLI, there are no pre-selected values, since users only type; hence we can go for it.

**Comment by VMois** [comment]
WRT2

> when I used --all, I get also deleted runs.

Without `--all` you also get deleted runs.

> The new behavior would be: hide deleted runs unless --show-deleted-runs or -all` is used.

Okay.

**Comment by tiborsimko** [comment]
> Without --all you also get deleted runs.

Yes, that's exactly the purpose:

```console
$ reana-client list --help | grep deleted
  --all                     Show all workflows including deleted ones.
```

It is just a compatibility option. For very old REANA releases, deleted runs were not shown unless you specified `--all`. Later we switched to always showing deleted runs, and we kept `--all` for backwards compatibility. Now when you add a new option `--show-deleted-runs`, the `--all` will behave just like it... That's why I mentioned that we are "seemingly bringing the old `reana-client` behaviour back" by not showing deleted runs unless one uses an option.
